### ISSUE TITLE
`EmberServerWebSocketSuite.respects withFilterPingPongs(false)` timeout flake

### ISSUE THREAD

**Issue by armanbilge** [http4s/http4s#6423]
```
==> X org.http4s.ember.server.EmberServerWebSocketSuite.respects withFilterPingPongs(false)  30.005s java.util.concurrent.TimeoutException: Future timed out after [30 seconds]
```

https://github.com/http4s/http4s/runs/6638446351?check_suite_focus=true#step:13:4004

**Comment by FrancescoSerra** [comment]
I didn't know munit was so heavily Future-based.
The only way to resolve this issue would seem to be to use `cats.effect.testkit.TestControl` ?

**Comment by armanbilge** [comment]
Yes, `Future` is the universal type that all Scala async stacks interop with :)

I don't think we can use `TestControl` for this one, since the test is based an asynchronous TCP socket and an external web socket client.

It's possible this hang is related to https://github.com/http4s/http4s/issues/6420#issuecomment-1138864091.

**Comment by armanbilge** [comment]
Flaked again in https://github.com/http4s/http4s/runs/7415071669?check_suite_focus=true#step:13:4069.
